com.unity.render-pipelines.high-definition/Runtime/Lighting/VolumetricLighting/VolumetricLighting.cs

```diff
@@ -261,9 +261,10 @@ static internal Vector3Int ComputeVolumetricViewportSize(HDCamera hdCamera, ref
             float screenFraction = controller.screenResolutionPercentage.value * 0.01f;
             int   sliceCount     = controller.volumeSliceCount.value;
 
-            int w = Mathf.RoundToInt(viewportWidth  * screenFraction);
-            int h = Mathf.RoundToInt(viewportHeight * screenFraction);
-            int d = sliceCount;
+            const int kMax3DTextureSize = 2048; // TODO: There's no SystemInfo.maxTextureSize equivalent for 3D textures
```

Comment:
So it's bad but I we need to solve it with adding SystemInfo.maxTextureSize for 3D texture as you said. So we should at least put a warning here Debug.Warning() to inform there is an issue with the same (and that it is a limit on metal platform).

Reply:
This has nothing to do with Metal, it's a generic limitation on DX11 as well. When using volumetrics dynamic resolution slider, you will easily go over the supported limit on any target. We've talked about this with @EvgeniiG before, besides checking for the maximum resolution, maybe there's some improvements that could be made into the actual dynamic resolution scaler itself? Now i was able to hit the resolution limits around 70% of the slider